core/src/main/java/org/springframework/security/oauth2/server/authorization/AccessTokenOAuth2Authorization.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.server.authorization;
+
+import org.springframework.lang.Nullable;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.util.Assert;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * An {@link OAuth2Authorization} implementation with {@link OAuth2AccessToken} access token.
+ *
+ * @author Krisztian Toth
+ */
+public class AccessTokenOAuth2Authorization extends OAuth2Authorization {
```

Comment:
Please remove this class as `OAuth2AccessToken` should be a member in `OAuth2Authorization`